File: app/src/main/java/com/awave/apps/droider/Feed/Interactors/FeedItemTouchHelper.java
package com.awave.apps.droider.Feed.Interactors;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;

//TODO Сделать сохранение статей свайпом
public class FeedItemTouchHelper extends ItemTouchHelper.Callback {
    private static final String TAG = "FeedItemTouchHelper";

    private OnFeedItemTouch mFeedItemTouch;

    public FeedItemTouchHelper(OnFeedItemTouch feedItemTouch) {
        mFeedItemTouch = feedItemTouch;
    }

    @Override
    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
        int swipeCards = ItemTouchHelper.START | ItemTouchHelper.END;
        return makeMovementFlags(0, swipeCards);
    }

    @Override
    public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
        mFeedItemTouch.onItemMove(viewHolder.getAdapterPosition(), target.getAdapterPosition());
        return true;
    }

    @Override
    public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
        mFeedItemTouch.onItemDismiss(viewHolder.getAdapterPosition());
    }

    @Override
    public boolean isLongPressDragEnabled() {
        return true;
    }

    @Override
    public boolean isItemViewSwipeEnabled() {
        return true;
    }
}

File: app/src/main/java/com/apps/wow/droider/Feed/Interactors/FeedItemTouchHelper.kt
package com.apps.wow.droider.Feed.Interactors

import android.support.v7.widget.RecyclerView
import android.support.v7.widget.helper.ItemTouchHelper

//TODO Сделать сохранение статей свайпом
class FeedItemTouchHelper(private val mFeedItemTouch: OnFeedItemTouch) : ItemTouchHelper.Callback() {

    override fun getMovementFlags(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder): Int {
        val swipeCards = ItemTouchHelper.START or ItemTouchHelper.END
        return ItemTouchHelper.Callback.makeMovementFlags(0, swipeCards)
    }

    override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
        mFeedItemTouch.onItemMove(viewHolder.adapterPosition, target.adapterPosition)
        return true
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
        mFeedItemTouch.onItemDismiss(viewHolder.adapterPosition)
    }

    override fun isLongPressDragEnabled(): Boolean {
        return true
    }

    override fun isItemViewSwipeEnabled(): Boolean {
        return true
    }
}
